06.bowling_object/frame.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Frame
+  attr_reader :frame_number, :shot_scores
+
+  def initialize(frame_number)
+    @frame_number = frame_number
+    @shot_scores = []
+    @bonus_scores = []
+  end
+
+  def record_shot(shot_score)
+    @bonus_scores << shot_score if (strike? && @bonus_scores.size < 2 || spare? && @bonus_scores.empty?) && frame_number < 9
```

Comment:
```suggestion
    @bonus_scores << shot_score if hogehoge?
```

ちょっと横に長くて読みづらいので抽象化するメソッドを用意したいかも

Reply:
`need_bonus?`メソッドを作成して対応しました